wazo_agid/modules/callerid_forphones.py

```diff
@@ -38,21 +40,82 @@ def callerid_forphones(agi: agid.FastAGI, cursor: DictCursor, args: list[str]) -
             user_uuid = callee_info.user_uuid
 
         tenant_uuid = agi.get_variable('WAZO_TENANT_UUID')
-        # It is not possible to associate a profile to a reverse configuration in the web
-        lookup_result = dird_client.directories.reverse(
-            profile='default',
-            user_uuid=user_uuid,
-            exten=cid_number,
-            tenant_uuid=tenant_uuid,
-        )
-        if lookup_result['display'] is not None:
-            logger.debug(
-                'Found caller ID from reverse lookup: "%s"<%s>',
-                lookup_result['display'],
-                cid_number,
+        numbers = [cid_number]
+        country = None
+        try:
+            tenant = objects.Tenant(agi, cursor, tenant_uuid)
+            country = tenant.country
+        except Exception as e:
+            msg = f'Could not fetch tenant: {e}'
+            logger.info(msg)
+            agi.verbose(msg)
+
+        if country:
+            parsed_number = phonenumbers.parse(cid_number, country)
```

Comment:
what happens if there's a parsing error? For ex., if the number is incompatible with the tenant country?

Reply:
Good point, I will add a `try... except` to handle that.